crates/nu-command/src/filters/split_by.rs

```diff
@@ -202,33 +202,24 @@ pub fn data_split(
                     }
                 }
                 _ => {
-                    return Err(ShellError::GenericError(
-                        "unsupported input".into(),
-                        "requires a table with one row for splitting".into(),
-                        Some(span),
-                        None,
-                        Vec::new(),
-                    ))
+                    return Err(ShellError::OnlySupportsThisInputType {
+                        exp_input_type: "Record".into(),
+                        wrong_type: v.get_type().to_string(),
+                        dst_span,
+                        src_span: v.span(),
+                    })
                 }
             }
         }
-        _ => {
-            return Err(ShellError::GenericError(
-                "unsupported input".into(),
-                "requires a table with one row for splitting".into(),
-                Some(span),
-                None,
-                Vec::new(),
-            ))
-        }
+        _ => return Err(ShellError::PipelineEmpty { dst_span }),
```

Comment:
If I read the code leading to this point correctly this could still be more than `PipelineData::Empty` e.g. an `ExternalStream`, right?

Reply:
Yes. In 84661a5c20a7fe3c604d80562488513e2e2c2ded I split `PipelineData::Empty` out from the other types which return a `PipelineMismatch`. I looked around and this seems most appropriate as `merge` treats a `ListStream` as a `List` and `ExternalStream` currently can't produce a `Record`.